tools/integration_tests/managed_folders/admin_permissions_test.go

```diff
@@ -33,18 +35,163 @@ import (
 // Boilerplate
 // //////////////////////////////////////////////////////////////////////
 
+const (
+	CreateTestFile = "createTestFile"
+)
+
 var (
-	bucket  string
-	testDir string
+	bucket           string
+	testDir          string
+	serviceAccount   string
+	localKeyFilePath string
 )
 
 type managedFoldersAdminPermission struct {
+	iamPermission    string
+	bucketPermission string
 }
 
 func (s *managedFoldersAdminPermission) Setup(t *testing.T) {
+	bucket, testDir = setup.GetBucketAndObjectBasedOnTypeOfMount(TestDirForManagedFolderTest)
+	createDirectoryStructureForNonEmptyManagedFolders(t)
+	if s.iamPermission != "nil" {
+		providePermissionToManagedFolder(bucket, path.Join(testDir, ManagedFolder1), serviceAccount, s.iamPermission, t)
+		providePermissionToManagedFolder(bucket, path.Join(testDir, ManagedFolder2), serviceAccount, s.iamPermission, t)
+	}
 }
 
 func (s *managedFoldersAdminPermission) Teardown(t *testing.T) {
+	revokePermissionToManagedFolder(bucket, path.Join(testDir, ManagedFolder1), serviceAccount, s.iamPermission, t)
+	revokePermissionToManagedFolder(bucket, path.Join(testDir, ManagedFolder2), serviceAccount, s.iamPermission, t)
+	cleanup(bucket, testDir, serviceAccount, s.iamPermission, t)
+}
+
+func (s *managedFoldersAdminPermission) TestCreateObjectInManagedFolder(t *testing.T) {
+	testDirPath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, ManagedFolder1)
+	file := path.Join(testDirPath, CreateTestFile)
+
+	createFileForTest(file, t)
+}
+
+func (s *managedFoldersAdminPermission) TestDeleteObjectInManagedFolder(t *testing.T) {
+	filePath := path.Join(setup.MntDir(), TestDirForManagedFolderTest, ManagedFolder1, FileInNonEmptyManagedFoldersTest)
+
+	err := os.Remove(filePath)
+	if err != nil {
+		t.Errorf("Error in removing file from managed folder: %v", err)
+	}
+}
+
+// Managed folders will not be deleted, but they will become empty. Default empty managed folders will be hidden.
```

Comment:
By Default empty managed folders will be hidden - Can we make a stat call to validate this?